be/src/io/cache/block_file_cache_downloader.cpp

```diff
@@ -229,6 +230,7 @@ void FileCacheBlockDownloader::download_segment_file(const DownloadFileMeta& met
         // TODO(plat1ko):
         //  1. Directly append buffer data to file cache
         //  2. Provide `FileReader::async_read()` interface
+        meta.ctx.is_dryrun = true;
```

Comment:
why meta.is_dryrun is not set by the caller?

Reply:
too many of them and is safer to set here.

Reply:
I'll set it by caller and use DCHECK here to ensure safety